dlib/dnn/core.h

```diff
@@ -3206,25 +3206,25 @@ namespace dlib
             unsigned long count_outputs() const
             {
                 if (subnetwork)
-                    return subnetwork->count_outputs() + output.size();
+                    return static_cast<unsigned long>(subnetwork->count_outputs() + output.size());
                 else
-                    return output.size();
+                    return static_cast<unsigned long>(output.size());
```

Comment:
In these kinds of cases, it might often make sense to change the function signature instead. But I wanted to keep the changes to a bare minimum, at least for the time being.